service/history/replicatorQueueProcessor.go

```diff
@@ -193,7 +193,8 @@ func (p *replicatorQueueProcessorImpl) processSyncActivityTask(task *persistence
 		},
 	}
 
-	return p.replicator.Publish(replicationTask)
+	retError = p.replicator.Publish(replicationTask)
```

Comment:
this change does nothing

Reply:
Actually you are right. Returning error will also update it for defer func(){}() 
The only problem can cause it is in processActivityRetryTimer